web-registry/src/lib/wallet/hooks/useLogin.tsx

```diff
@@ -0,0 +1,58 @@
+import { useCallback } from 'react';
+import axios from 'axios';
+
+import { UseStateSetter } from 'types/react/use-state';
+import { apiUri } from 'lib/apiUri';
+import { SignArbitraryType, Wallet } from 'lib/wallet/wallet';
+
+type Params = {
+  signArbitrary?: SignArbitraryType;
+  setError: UseStateSetter<unknown>;
+  setAccountId: UseStateSetter<string | undefined>;
+};
+
+export const useLogin = ({ signArbitrary, setError, setAccountId }: Params) => {
+  const login = useCallback(
+    async (wallet?: Wallet): Promise<void> => {
+      try {
+        if (wallet?.address && signArbitrary) {
+          const instance = axios.create({
```

Comment:
Do we really need to use Axios here?
Since we have low usage of it, I wanted to remove it from our bundle eventually.

Reply:
it's quite nice to just use the axios instance so we don't have to repeat the url and credentials params for every requests, but I'd be fine just using fetch too, although the code might become a bit less readable, unless we define some wrapper function around it

Reply:
Ok, let's keep it then for now.